### PR TITLE
Make arb-validator use a generic efficient Machine interface

### PR DESCRIPTION
**Pull request by hkalodner**
This repo now uses a simplified `Machine` interface declared in https://github.com/OffchainLabs/arb-util and removes dependency on https://github.com/OffchainLabs/arb-avm other than when creating an actual machine.

### CODE REVIEW

File: ethvalidator/coordinator.go [hunk 1]
```diff
@@ -322,17 +323,19 @@ type ValidatorCoordinator struct {
...
 func NewCoordinator(
 	name string,
-	machine *vm.Machine,
+	machine machine.Machine,
 	key *ecdsa.PrivateKey,
 	config *valmessage.VMConfiguration,
 	challengeEverything bool,
 	connectionInfo ethbridge.ArbAddresses,
 	ethURL string,
+	maxSteps int32,
```
Comment (ughe) [inline ethvalidator/coordinator.go]: Why are the max steps an `int32`? Would using Ethereum's `U256` work?

File: cmd/coordinatorServer/main.go [hunk 2]
```diff
@@ -59,6 +61,10 @@ func attachProfiler(router *mux.Router) {
 // 4) Global EthBridge addresses json file
 // 5) ethURL
 func main() {
+	// cFilename := C.CString("contract.ao")
+	// C.machine_create(cFilename)
+	// C.free(unsafe.Pointer(cFilename))
```
Comment (ughe) [inline cmd/coordinatorServer/main.go]: Should this information include a comment i.e. `// DEBUG`

File: validator/validator.go [hunk 3]
```diff
@@ -165,6 +169,8 @@ func (validator *Validator) CloseUnanimousAssertionRequest() <-chan bool {
 	return resultChan
 }
 
+const maxCallSteps int32 = math.MaxInt32
```
Comment (ughe) [inline validator/validator.go]: This is declared once and used once. Most other files just use `math.MaxInt32` for the same purpose.

File: core/core.go [hunk 4]
```diff
@@ -45,129 +44,77 @@ func (c *Config) GetConfig() *Config {
...
 		timeBounds,
-		inbox.Receive(),
 	)
-	newState.RunUntilStop()
-	assDef := assCtx.Finalize(newState)
-
-	newAssertion := assDef.GetAssertion()
-	newBalance := c.balance.Clone()
-	// This spend is guaranteed to be correct since the VM made sure to only produce on outgoing if it could spend
-	_ = newBalance.SpendAll(protocol.NewBalanceTrackerFromMessages(newAssertion.OutMsgs))
```
Comment (ughe) [inline core/core.go]: Why is the `SpendAll` removed? Is it related to `SendOffchainMessages`?

File: cmd/challengeTest/main.go [hunk 5]
```diff
@@ -96,7 +98,15 @@ func main() {
 
 	ethURL := os.Args[3]
 
-	coordinator, err := ethvalidator.NewCoordinator("Alice", machine.Clone(), key1, config, false, connectionInfo, ethURL)
+	coordinator, err := ethvalidator.NewCoordinator(
+		"Alice",
+		machine.Clone(),
+		key1, config,
```
[the remark this answers is not in the document]
Comment (ughe) [reply cmd/challengeTest/main.go]: Fixed.

File: ethvalidator/coordinator.go (hunk 1 above)
Comment (hkalodner) [reply ethvalidator/coordinator.go]: We should change this to uint64 in a future commit

File: cmd/coordinatorServer/main.go (hunk 2 above)
Comment (hkalodner) [reply cmd/coordinatorServer/main.go]: I'll remove this

File: validator/validator.go (hunk 3 above)
Comment (hkalodner) [reply validator/validator.go]: Removed

**Pull request merged**